### ISSUE TITLE
Returning a model with aliases from an endpoint breaks validation

### ISSUE THREAD

**Issue by ghost** [tiangolo/fastapi#1070]
### Describe the bug

I believe this is somewhat related to https://github.com/tiangolo/fastapi/issues/1063 and also not quite sure if this should be a feature or a bug, but it is causing some trouble for me anyway.

Basically I would like to use the same functions that are defined as FastAPI endpoints for other use cases as well. For this I would like to have them all return Pydantic models instead of dicts to have the typing in place. However, if any of the models has aliases the validation simply breaks, everything seems to work fine without validation.

### To Reproduce

1. Create a file with:

```Python
from fastapi import FastAPI
from pydantic import BaseModel, Field

class SimpleModel(BaseModel):
    name: str
    value: int

class AliasedModel(BaseModel):
    name: str = Field(..., alias="aliased_name")
    value: int

app = FastAPI()

@app.get("/simple", response_model=SimpleModel, response_model_by_alias=True)
def read_simple():
    return SimpleModel(name="name", value=1234)

@app.get("/aliased", response_model=AliasedModel, response_model_by_alias=True)
def read_aliased():
    return AliasedModel(name="name", value=1234)
```

3. Open the browser and call the endpoint `/simple`.
4. It returns a JSON with `{"name": "name", "value": 1234}`.
3. Open the browser and call the endpoint `/aliased`.
4. It returns an Internal Server Error
5. But I expected it to return `{"aliased_name": "name", "value": 1234}`.

### Expected behavior

I would expect the response to be valid, since it is a valid instance of the given `response_model` class. It feels very counter-intuitive that returning a model from the endpoint instead of a dict suddenly fails.

### Environment

- OS: Linux
- FastAPI 0.49.0
- Python 3.7.6:

**Comment by johnanthonyowens** [comment]
This looks like another manifestation of #1043

**Comment by juhovh-aiven** [comment]
You're right, sorry about that. I'll close this and let's keep your report instead, it is definitely the same problem.

**Comment by tiangolo** [comment]
Thanks for the discussion here! :cake: 

And thanks @juhovh-aiven for reporting back and closing the issue. (And for the fix :tada: )
